docs/decisions/0019-scaling-styles-with-design-tokens.rst

```diff
@@ -0,0 +1,261 @@
+^^^^^^^^^^^^^^^^
+
+An exhaustive representation of every value associated with specific components in the design system. Component tokens give explicit control over component-specific styles. They may inherit from other token types.
+
+Example::
+
+  {
+    "color": {
+      "btn": {
+        "bg": {
+          "primary": {
+            "value": "{color.theme.default.primary.value}"
+          }
+        }
+      }
+    }
+  }
+
+The above component token would be transformed to a CSS variable by ``style-dictionary`` as follows::
+
+  --pgn-color-btn-bg-primary: var(--pgn-color-theme-default-primary);
+
+By creating tokens within these 3 categories, the Paragon theming system provides more explicit control in how design tokens are utilized throughout the design system and its components.
+
+For example, by using alias tokens, theme authors may change the style properties of components consuming a specific alias token(s) without needing to necessarily change any underlying global token itself.
+
+In the token examples shown above, say the button component token referred directly to a global token instead. To update that button's background color (say, to a darker shade of the base primary color), the theme author must update the foundational global token to change the button color. However, this may have unintentend consequences in changing the color of components that were not intended to be changed.
+
+Instead, by having the component token inherit from an alias token, theme authors can modify the alias token without needing to change the underlying global token, which helps mitigate concerns around changing a foundational style property heavily used throughout the entire design system.
+
+In the above example, for instance, the value of ``--pgn-color-theme-default-primary`` could be changed to ``{color.primary.700}`` rather than changing the underlying value of ``{color.primary.500}`` directly, which might need to remain at its current value given its use elsewhere throughout the design system.
+
+Additional token file examples
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Please refer to the `tokens <https://github.com/openedx/paragon/tree/alpha/tokens/src>`__ defined in the current ``alpha`` release of Paragon.
+
+Implications for theme authors (e.g., ``@edx/brand`` packages)
+------------------------------------------------------------
+
+Currently, Paragon recommends theme authors to create a theme package such as ``@edx/brand-openedx`` (`Github <https://github.com/openedx/brand-openedx>`__) and ``@edx/brand-edx.org`` (`Github <https://github.com/edx/brand-edx.org>`__).
+
+While the migration from SCSS variables to CSS variables is a breaking change for theme authors, we have tried to mitigate this by keeping the existing SCSS variables but defining them such that their values refer to the new CSS variables. Because SCSS can't evaluate the CSS variable at runtime, it utilizes the CSS variable in the resulting output CSS used in the browser.
```

Comment:
Part of me feels like it might be better to "rip the band-aid off" and directly use css vars everywhere scss vars were being used. My main worry is having a mix of scss vars and css vars scattered throughout stylesheets preventing a true "source of truth" for a value.

For example, let's say we have `my-scss-var-foo` set to `my-css-var-foo`:
* if all the stylesheets use `my-scss-var-foo`, and never use `my-css-var-foo` directly, changing `my-scss-var-foo` to be `my-css-var-bar` would work as expected.
* howerver, if some stylesheets use `my-scss-var-foo` and some use `my-css-var-foo`, changing `my-scss-var-foo` could lead to unexpected behavior.

I can think of 2 possible ways to prevent this problem:
* Watch PRs that change scss vars like a hawk, don't let it happen
* Only use css vars

I know there was probably a lot of discussion/thought put into this decision, I'd love to hear any thoughts (or if there was a pros/cons list) that came up when building towards this strategy.

Reply:
@brian-smith-tcril I agree, this should likely be revisited. We did "rip the b(r)and-aid off" (pardon the terrible pun with the (r) haha!) in `@edx/paragon` itself by updating all the general theme and component styles to no longer use the SCSS variable in favor of using the CSS variable directly (e.g., see `Alert` [styles](https://github.com/openedx/paragon/blob/cc022bb0b071101d0bc3a6392f0ec69f5b874e5d/src/Alert/Alert.scss#L13)).

However, Paragon also does still have the `_variables.scss` file around in the `alpha` release, where each of the SCSS variables point to the associated CSS variable. I agree this isn't ideal for maintainability/understandability (e.g., single source of truth). Would you agree that we should likely delete `_variables.scss` in the design tokens release, if possible, at this point? Consumers should no longer be importing anything from Paragon's SCSS directly anymore but would need to verify that the `build-scss` script in Paragon doesn't rely on `_variables.scss`.